crates/kitsune_p2p/fetch/src/pool.rs

```diff
@@ -329,280 +413,56 @@ pub struct FetchPoolItem {
     size: Option<RoughInt>,
     /// Opaque user data specified by the host
     pub context: Option<FetchContext>,
-    /// The last time we tried fetching this item from any source
-    last_fetch: Option<Instant>,
+    /// If there is a response pending for this item then track the source and when the request was made.
+    pending_response: Option<PendingItemResponse>,
 }
 
+/// Tracks the source and when a request was made for a [`FetchPoolItem`]. This is used to track timeouts
+/// for sources that don't respond before the configured timeout.
 #[derive(Debug, PartialEq, Eq)]
-struct SourceRecord {
+pub struct PendingItemResponse {
+    when: Instant,
     source: FetchSource,
-    transfer_method: TransferMethod,
-    last_request: Option<Instant>,
-}
-
-impl SourceRecord {
```

Comment:
No longer required, the requests are tracked against the fetch pool item. That lets us track timeouts by source and also maintain information about those timeouts against the source, rather than per item